core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -324,14 +334,14 @@ contract PricelessPositionManager is FeePayer {
 
     /**
      * @notice After a contract has passed maturity all token holders can redeem their tokens for underlying at
-     * the prevailing price defined by the DVM from the `expire` function. 
+     * the prevailing price defined by the DVM from the `expire` function.
      * @dev This Burns all tokens from the caller of `tokenCurrency` and sends back the proportional amount of `collateralCurrency`.
      */
     function settleExpired() external onlyPostExpiration() fees() {
         // Get the current settlement price and store it. If it is not resolved will revert.
-        if (!hasExpiryPrice) {
+        if (contractState != ContractState.ExpiredPriceReceived) {
```

Comment:
So if contractState is `Active` then this method will succeed but do nothing.

Also, if  contractState `ExpiredPriceReceived `, then the contract does nothing. 

But if contractState is `ExpiredPriceRequested` and the DVM hasn't resolved a price yet-- then I assume `_getOraclePrice()` would revert, but we should handle this "do-nothing" result consistently.

We could add a modifier `onlyPriceRequestedState()` to ensure that this method either succeeds (sets an `expiryPrice` and changes contractState to `ExpiredPriceReceived`) or throws.

Let me know if I'm misunderstanding

Reply:
This if statement is only used to set the `expiryPrice` and `contractState` variables iff the contract time is post expiration via the `onlyPostExpiration` modifier.

If the contract state is in `ContractState.ExpiredPriceReceived` then the if statement is passed and the rest of settleExpired works as expected meaning that withdrawals work in the normal way. this is identical to the logic we had before with the `hasExpiryPrice` bool. 

This _could_ be done with a modifier but this is the only instance wherein this is used so it's probably better to leave it like this I think. Happy to discuss more :)

Reply:
Ok I forgot about the rest of `settleExpired` haha. 

But what happens if someone tries to call this when contract state is `Open` but post-expiry? i.e. no one has called `expire` yet

Reply:
this is a good point. Right now what will happen is the function will go through no problems because the `onlyPostExpiration` will not block the call. However the `expiryPrice` will be 0 which will result in no payout value. We could add a modifier for something like `onlyExpired` that checks that the `contractState != ContractState.Open`.

Reply:
I think we should revert it, which would be the same result as if `ContractState` is `ExpiredPriceRequested` but the price has not been received yet. 

This way, if there is no price, then `settleExpired` reverts

Reply:
I agree. I will add an additional modifier to this and add to the test coverage.